source/internal.d.ts

```diff
@@ -13,7 +14,34 @@ Infer the length of the given array `<T>`.
 
 @link https://itnext.io/implementing-arithmetic-within-typescripts-type-system-a1ef140a6f6f
 */
-type TupleLength<T extends readonly unknown[]> = T extends {readonly length: infer L} ? L : never;
+type ArrayLength<T extends readonly unknown[]> = T extends {readonly length: infer L} ? L : never;
+
+/**
+Infer the length of the given tuple `<T>`.
+
+Returns `never` if the given type is an non-fixed-length array like `Array<string>`.
```

Comment:
Isn't `non-fixed-length` just another name for a tuple?

Reply:
> Isn't `non-fixed-length` just another name for a tuple?

tuple length is fixed, array length is non-fixed. I want to emphasise that this array is non-fixed and not a tuple.

Reply:
Ok